include/url.h

```diff
@@ -79,6 +79,11 @@ namespace Url
          */
         Url& abspath();
 
+        /**
+         * Evaluate 'other' relative to this URL, placing the result in this object.
+         */
+        Url& relative(const std::string& other);
```

Comment:
FWIW, I tend to prefer *non-mutating* members, but it's easy to make a non-mutating version from a mutating version. Going the other way is less efficient.